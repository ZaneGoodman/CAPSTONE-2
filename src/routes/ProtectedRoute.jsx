import { Navigate, Outlet } from "react-router-dom";
import React from "react";
import { useAuth } from "../provider/authProvider";
//Keep watch over each protected route, redirect to login page if not authenticated
export const ProtectedRoute = () => {
  const { token } = useAuth();
  console.log("protected route:", token);
  if (!token) {
    return <Navigate to="/login" />;
  }

  return <Outlet />;
};
