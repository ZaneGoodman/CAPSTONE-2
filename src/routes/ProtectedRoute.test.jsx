import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import AuthProvider from "../provider/authProvider";
import { ProtectedRoute } from "./ProtectedRoute";

test("renders Routes component with authentication", async () => {
  render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <ProtectedRoute />
      </MemoryRouter>
    </AuthProvider>
  );
});

//Snapshot test  <MemoryRouter>
it("Routes matches snapshot", async () => {
  const { asFragment } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <ProtectedRoute />
      </MemoryRouter>
    </AuthProvider>
  );
  await expect(asFragment()).toMatchSnapshot();
});
