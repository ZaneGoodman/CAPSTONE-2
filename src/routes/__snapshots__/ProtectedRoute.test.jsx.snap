// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes matches snapshot 1`] = `<DocumentFragment />`;
